pkg/controllers/nodeclass/status/controller.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+
+	"github.com/awslabs/operatorpkg/status"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/karpenter-provider-cluster-api/pkg/apis/v1alpha1"
+	"sigs.k8s.io/karpenter/pkg/operator/injection"
+)
+
+type Controller struct {
+	kubeClient client.Client
+}
+
+func NewController(kubeClient client.Client) *Controller {
+	return &Controller{
+		kubeClient: kubeClient,
+	}
+}
+
+func (c *Controller) Reconcile(ctx context.Context, nodeClass *v1alpha1.ClusterAPINodeClass) (reconcile.Result, error) {
+	ctx = injection.WithControllerName(ctx, "nodeclass.status")
+	stored := nodeClass.DeepCopy()
+
+	// for now, we just want to set all nodeclasses to ready. in the future we may want
+	// to add other types of checks to ensure that the underlying cluster-api objects are
+	// also ready.
+	ready := nodeClass.StatusConditions().Get(status.ConditionReady)
+	if ready.IsUnknown() || ready.IsFalse() {
```

Comment:
I was worried about "what happens if the condition doesn't exist at all", but it looks like `Get()`  will return nil if it's not there, and `IsUnknown()` does regard nil as unknown, so that's perfect.